src/libraries/System.Private.CoreLib/src/System/IO/Strategies/OSFileStreamStrategy.cs

```diff
@@ -129,9 +104,6 @@ internal sealed override SafeFileHandle SafeFileHandle
                     FileStreamHelpers.Seek(_fileHandle, _filePosition, SeekOrigin.Begin);
                 }
 
-                _lengthCanBeCached = false;
```

Comment:
When we have introduced file length caching for the first time, we were caching it also when the file was opened for writing. We have tried to update the cached length in appropriate places, but at the end we have decided that it's not worth it (mostly due to the fact that users can start multiple concurrent write operations).

File caching is allowed only on Windows, when the file has not been opened for writing and when it has not been shared with others for writing.

Since it's impossible to use such file handle to modify the file, I've decided to remove the file length cache invalidaiton from `FileStreamStrategy.SafeFileHandle` (called by `FileStream.SafeFileHandle`).